webdriver/tests/bidi/storage/set_cookie/partition.py

```diff
@@ -1,34 +1,46 @@
 import pytest
 from webdriver.bidi.modules.storage import BrowsingContextPartitionDescriptor, StorageKeyPartitionDescriptor
 from .. import assert_cookie_is_set, create_cookie
+from ... import recursive_compare
 
 pytestmark = pytest.mark.asyncio
 
 
-async def test_partition_context(bidi_session, set_cookie, top_context, test_page, origin, domain_value):
+def assert_set_cookie_result(set_cookie_result, partition):
+    """
+    Asserts the result of `set_cookie` command depending on the partition type.
+    """
+    if isinstance(partition, BrowsingContextPartitionDescriptor):
+        # Browsing context does not require a `sourceOrigin` partition key, but it can be present depending on the
+        # browser implementation.
+        # `recursive_compare` allows the actual result to be any extension of the expected one.
+        recursive_compare({'partitionKey': {}, }, set_cookie_result)
+        return
+    if isinstance(partition, StorageKeyPartitionDescriptor):
```

Comment:
I think it should also check is partition has sourceOrigin as it is optional.

Reply:
In `StorageKeyPartitionDescriptor` the `sourceOrigin ` is not optional, but required. And we check it.

If you mean the `BrowsingContextPartitionDescriptor`, the result can contain not only the `sourceOrigin`, but any other browser-specific field.